### PR TITLE
PIMOB-1379: Improve cardholder input view behaviour

### PR DESCRIPTION
**Pull request by alex-ioja-yang-cko**
## Proposed changes

[Jira task](https://checkout.atlassian.net/browse/PIMOB-1379)
Improve handling of name input validation.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [X] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

Changes were needed due to a different location updating a field that by default would start empty. Used the same pattern as all other communication paths between viewModel and viewController.
Changed default names provided to cater for more extensive alphabet.

### CODE REVIEW

File: Source/UI/CommonUI/View/Component/InputView.swift [hunk 1]
```diff
@@ -89,6 +89,10 @@ class InputView: UIView {
         updateTextFieldContainer(style: style)
         textFieldView.update(with: style.textfield)
         updateErrorView(style: style)
+
+        if let textFieldValue = textFieldValue {
+            textFieldView.textField.text = textFieldValue
+        }
```
Comment (ehab-al-cko) [inline Source/UI/CommonUI/View/Component/InputView.swift:95]: UI elements should not be overridden from another class. i suggest to set style.text
You may update it when update the style in  `CardHolderView` 
`func update(style: CellTextFieldStyle, text: String?)`
 ```suggestion
```

File: Source/UI/PaymentForm/View/CardholderName/CardholderView.swift [hunk 2]
```diff
@@ -30,9 +37,9 @@ final class CardholderView: UIView {
         fatalError("init(coder:) has not been implemented")
     }
 
-    func update(style: CellTextFieldStyle) {
+    func update(style: CellTextFieldStyle, text: String?) {
         self.style = style
-        cardholderInputView.update(style: style)
+        cardholderInputView.update(style: style, textFieldValue: text)
```
Comment (ehab-al-cko) [inline Source/UI/PaymentForm/View/CardholderName/CardholderView.swift:42]: I would suggest to override the text here 
```suggestion
       if let text = text {
            style.textfield.text = text
        }
        cardholderInputView.update(style: style)
```

File: Source/UI/CommonUI/View/Component/InputView.swift (hunk 1 above)
Comment (alex-ioja-yang-cko) [reply Source/UI/CommonUI/View/Component/InputView.swift:95]: Absolutely agree and to be honest this is why I would have liked the UI properties to always be private. Just thought if they're already internal, will use them...

Will change as I do agree with the point you raised ➕

File: Source/UI/PaymentForm/View/CardholderName/CardholderView.swift (hunk 2 above)
Comment (alex-ioja-yang-cko) [reply Source/UI/PaymentForm/View/CardholderName/CardholderView.swift:42]: Cool observation, I knew I must have missed something 🤦‍♂️

**Pull request merged**